R/rules.R

```diff
@@ -295,3 +294,93 @@ sv_between <- function(left,
     }
   }
 }
+
+#' Validate that a field is part of a defined set
+#'
+#' The `sv_in_set()` function checks whether the field value is part of a
+#' specified set of values.
+#' 
+#' @param set A vector of elements for which the field value must be a part of
+#'   to pass validation. To allow an empty field, `NA` should be included in the
+#'   `set` vector.
+#' @param allow_nan If `FALSE` (the default), then the presence of a `NaN` value
+#'   element will cause validation to fail.
+#' @param message The validation error message to use if a value fails to match
+#'   the rule. By default, this is generic message provided by **shinyvalidate**
+#'   but a custom message can be provided here.
+#' @param msg_limit The limit of `set` values to include in the
+#'   automatically-generated error message (i.e., when `message = NULL`). If the
+#'   number of elements provided in `set` is greater than `msg_limit` then only
+#'   the first `<message_limit>` set elements will be echoed along with text
+#'   that states how many extra elements are part of the `set`.
+#'
+#' @return A function suitable for using as an
+#'   [`InputValidator$add_rule()`][InputValidator] rule.
+#'
+#' @examples
+#' # Ignore withReactiveDomain(), it's just required to get this example to run
+#' # outside of Shiny
+#' shiny::withReactiveDomain(shiny::MockShinySession$new(), {
+#' 
+#'   iv <- InputValidator$new()
+#' 
+#'   iv$add_rule("count", sv_in_set(1:5))
+#'   iv$add_rule("count", ~if (. <= 0) "A positive value is required")
+#'
+#' })
+#' @export
+sv_in_set <- function(set,
+                      allow_nan = FALSE,
```

Comment:
Let's lose `allow_nan`, people can just include `NaN` in the `set`

Reply:
Done.